cli/command/stack/deploy_composefile.go

```diff
@@ -121,11 +122,20 @@ func propertyWarnings(properties map[string]string) string {
 func getConfigDetails(composefile string) (composetypes.ConfigDetails, error) {
 	var details composetypes.ConfigDetails
 
-	absPath, err := filepath.Abs(composefile)
-	if err != nil {
-		return details, err
+	if composefile == "-" && runtime.GOOS != "windows" {
+		composefile = "/dev/stdin"
```

Comment:
I think we can make this easier to test, and supported on windows.

Instead of setting `composefile = "/dev/stdin"` here, only set WorkingDir.

You'll also need to pass `dockerCli.In()` as an `io.Reader` from `deployCompose()` where this is called, so we have access to it in `getConfigFile()`.

In `getConfigFile` before calling `ioutil.ReadFile()` check if `filename == "-"` and use this instead to get the bytes:

```go
bytes, err = ioutil.ReadAll(stdin)
```